### PR TITLE
Logging Cleanup - Part 5

### PR DESCRIPTION
**Pull request by xWink**
**What type of PR is this?**

Cleanup

**Which issue does this PR fix**:
#126 (final)

**What does this PR do / Why do we need it**:
- Removes many unwanted info logs
- Changes some error logs that do not actually indicate an error (e.g. expected errors that result in requeue) to debug
- Removes struct embedding and use of ambiguous %v in logs
- Add route reconciler info logs that match the other reconcilers
- General logging language/clarity improvements
- Added some missing error handling
- Overall, running this with only info level logs should be way easier to follow along, and running with debug logs should be much less noisy without losing relevant/useful context

**If an issue # is not available please add repro steps and logs from aws-gateway-controller showing the issue**:

**Testing done on this change**:

```
Ran 32 of 32 Specs in 2058.765 seconds
SUCCESS! -- 32 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestIntegration (2059.73s)
PASS
ok      github.com/aws/aws-application-networking-k8s/test/suites/integration   2060.453s
```

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this PR introduce any user-facing change?**:

No, only logging changes

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by zijun726911 — APPROVED** [review]
LGTM

**Comment by xWink** [comment]
Reran e2e tests after resolving merge conflicts
```
Ran 32 of 32 Specs in 2135.827 seconds
SUCCESS! -- 32 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestIntegration (2136.74s)
PASS
ok      github.com/aws/aws-application-networking-k8s/test/suites/integration   2137.530s
```

File: pkg/gateway/model_build_targetgroup.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (t *svcExportTargetGroupModelBuildTask) run(ctx context.Context) error {
 
 	err = t.BuildTargets(ctx)
 	if err != nil {
-		t.log.Errorf("Failed to build targets for service export %s-%s due to %w",
+		t.log.Debugf("Failed to build targets for service export %s-%s due to %s",
```
Comment (mikhail-aws) [inline pkg/gateway/model_build_targetgroup.go:97]: should be error log

File: pkg/gateway/model_build_lattice_service.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func (t *latticeServiceModelBuildTask) buildModel(ctx context.Context) error {
 	}
 
 	if err := t.buildTargetsForRoute(ctx); err != nil {
-		t.log.Errorf("failed to build targets due to %s", err)
+		t.log.Debugf("failed to build targets due to %s", err)
```
Comment (mikhail-aws) [inline pkg/gateway/model_build_lattice_service.go:89]: should be error log

**Pull request merged**